PubSub seems to fail to parse shoutout events

### Prerequisites

- [X] If this is a question/suggestion, I have already considered discussing it on the [Discord Server](https://discord.gg/FQ5vgW3)
- [x] I have attempted to look for similar github issues already

### Expected Behavior

My application create a twitch client with the PubSub feature.
When subscribing to a shoutout creation event of a channel, i should receive the shoutout creation event.
This event should be parsed then dispatched through the event manager (i'm using the SimpleEventManager) and pass it to my consumer so that i can process it.

### Current Behavior

The subscription seems to work and i receive the event with its content.
However, the pubsub service seems to fail to parse the message as you can see in the Stacktrace below.

### Log output / Stacktrace

```txt
2023-01-13 21:56:06.958  WARN 10684 --- [  ReadingThread] com.github.twitch4j.pubsub.TwitchPubSub  : PubSub: Unparsable Message: {"type":"MESSAGE","data":{"topic":"shoutout.135074820","message":"{\"type\":\"create\",\"data\":{\"broadcasterUserID\":\"135074820\",\"targetUserID\":\"42941889\",\"targetLogin\":\"arnaudryku\",\"targetUserProfileImageURL\":\"https://static-cdn.jtvnw.net/jtv_user_pictures/71cc5b31-6cc1-4761-b7c4-7a1fccfaa56a-profile_image-%s.jpeg\",\"sourceUserID\":\"72738644\",\"sourceLogin\":\"diatorker\",\"shoutoutID\":\"d2a6ac54-1d8a-4e44-8b5d-f7c1ee4938e6\",\"targetUserDisplayName\":\"ArnaudRyku\",\"targetUserCTAInfo\":\"\",\"targetUserPrimaryColorHex\":\"000000\"}}"}}
 - [java.lang.IllegalArgumentException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0] (through reference chain: com.github.twitch4j.pubsub.domain.CreateShoutoutData["targetUserCTAInfo"])]

java.lang.RuntimeException: java.lang.IllegalArgumentException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0] (through reference chain: com.github.twitch4j.pubsub.domain.CreateShoutoutData["targetUserCTAInfo"])
	at com.github.twitch4j.common.util.TypeConvert.convertValue(TypeConvert.java:56) ~[twitch4j-common-1.13.0.jar:1.13.0]
	at com.github.twitch4j.pubsub.TwitchPubSub.onTextMessage(TwitchPubSub.java:771) ~[twitch4j-pubsub-1.13.0.jar:1.13.0]
	at com.github.twitch4j.client.websocket.WebsocketConnection$1.onTextMessage(WebsocketConnection.java:123) ~[twitch4j-client-websocket-1.13.0.jar:1.13.0]
	at com.neovisionaries.ws.client.ListenerManager.callOnTextMessage(ListenerManager.java:353) ~[nv-websocket-client-2.14.jar:na]
	at com.neovisionaries.ws.client.ReadingThread.callOnTextMessage(ReadingThread.java:266) ~[nv-websocket-client-2.14.jar:na]
	at com.neovisionaries.ws.client.ReadingThread.callOnTextMessage(ReadingThread.java:244) ~[nv-websocket-client-2.14.jar:na]
	at com.neovisionaries.ws.client.ReadingThread.handleTextFrame(ReadingThread.java:969) ~[nv-websocket-client-2.14.jar:na]
	at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:752) ~[nv-websocket-client-2.14.jar:na]
	at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108) ~[nv-websocket-client-2.14.jar:na]
	at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64) ~[nv-websocket-client-2.14.jar:na]
	at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45) ~[nv-websocket-client-2.14.jar:na]
Caused by: java.lang.IllegalArgumentException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0] (through reference chain: com.github.twitch4j.pubsub.domain.CreateShoutoutData["targetUserCTAInfo"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4393) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4324) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.github.twitch4j.common.util.TypeConvert.convertValue(TypeConvert.java:54) ~[twitch4j-common-1.13.0.jar:1.13.0]
	... 10 common frames omitted
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0] (through reference chain: com.github.twitch4j.pubsub.domain.CreateShoutoutData["targetUserCTAInfo"])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:276) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty._throwAsIOE(SettableBeanProperty.java:623) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty._throwAsIOE(SettableBeanProperty.java:611) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:143) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:313) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:176) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4388) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	... 12 common frames omitted
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0]
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4765) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4667) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3597) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.github.twitch4j.common.util.TypeConvert.jsonToObject(TypeConvert.java:38) ~[twitch4j-common-1.13.0.jar:1.13.0]
	at com.github.twitch4j.pubsub.domain.CreateShoutoutData.unpackCallToAction(CreateShoutoutData.java:46) ~[twitch4j-pubsub-1.13.0.jar:1.13.0]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:141) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	... 15 common frames omitted
```


### Environment

- **Twitch4J v1.13.0**: 
- **IntelliJ IDEA 2022.1.2**: 
- **OpenJDK 11.0.9**: 
- **Gradle 7.3.1**: 
- **Windows 10 21H2**: 

Also, the application is hosted in a spring web app. Not sure if it could cause any problem...

### Build Configuration

```gradle
plugins {
    id 'org.springframework.boot' version '2.6.2'
    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
    id 'java'
    id "com.dorongold.task-tree" version "2.1.0"
    id 'org.kordamp.gradle.markdown' version '2.2.0'
    id 'ru.vyarus.mkdocs' version '2.3.0'
}

group = 'org.latuile'
version = '2.0.3'
archivesBaseName = 'StreamerTools'
sourceCompatibility = '11'

mkdocs.publish.docPath = ''

bootJar {
    archiveFileName = 'streamer-tools.jar'
    finalizedBy 'mkdocsBuild'
}

mkdocsBuild {
    finalizedBy 'copyDocs'
}

task copyDocs(type: Copy) {
    from "$buildDir/mkdocs/"
    into "$buildDir/libs/doc/"
    finalizedBy 'copyDists'
}

task copyDists(type: Copy) {
    from "$projectDir/zipped/"
    into jar.destinationDirectory
    finalizedBy 'createDistZip'
}

task createDistZip(type: Zip){
    from jar.destinationDirectory
    destinationDirectory = file("$projectDir/dist/")
}

configurations {
    compileOnly {
        extendsFrom annotationProcessor
    }
}

repositories {
    mavenCentral()
}

ext['log4j2.version'] = '2.17.1'
dependencies {
    implementation 'org.springframework.boot:spring-boot-starter-thymeleaf:2.6.7'
    implementation 'org.springframework.boot:spring-boot-starter-web:2.6.7'
    implementation 'org.springframework.boot:spring-boot-starter-websocket:2.6.7'
    implementation 'org.springframework.boot:spring-boot-starter-security:2.6.7'
    implementation 'org.springframework.boot:spring-boot-starter-data-jpa:2.6.7'
    implementation 'com.h2database:h2:2.1.212'
    implementation 'com.google.apis:google-api-services-youtube:v3-rev20220418-1.32.1'
    implementation 'com.google.http-client:google-http-client-jackson2:1.41.7'
    implementation 'org.webjars:webjars-locator-core:0.50'
    implementation 'org.webjars:sockjs-client:1.5.1'
    implementation 'org.webjars:stomp-websocket:2.3.4'
    implementation 'org.webjars:bootstrap:5.1.3'
    implementation 'org.webjars:jquery:3.6.0'
    implementation 'se.michaelthelin.spotify:spotify-web-api-java:7.0.1'
    implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.13.2'
    implementation 'com.fasterxml.jackson.core:jackson-databind:2.13.2.2'
    implementation 'io.jsonwebtoken:jjwt:0.9.1'
    implementation 'com.github.twitch4j:twitch4j:1.13.0'
    compileOnly 'org.projectlombok:lombok:1.18.24'
    developmentOnly 'org.springframework.boot:spring-boot-devtools:2.6.7'
    annotationProcessor 'org.projectlombok:lombok:1.18.24'
    testImplementation 'org.springframework.boot:spring-boot-starter-test:2.6.7'
}

test {
    useJUnitPlatform()
}
```

### Possible Solution

It seems that the field targetUserCTAInfo is an empty string in my case. It seems that the parser tries parse this as a sub JSON field but fails because it is empty

### Additional Context/Screenshots

_No response_

--- a/pubsub/src/main/java/com/github/twitch4j/pubsub/domain/CreateShoutoutData.java
+++ b/pubsub/src/main/java/com/github/twitch4j/pubsub/domain/CreateShoutoutData.java
@@ -43,6 +43,7 @@ public class CreateShoutoutData {
 
     @JsonProperty("targetUserCTAInfo")
     private void unpackCallToAction(String ctaInfo) {
+        if (ctaInfo == null || ctaInfo.isEmpty()) return;
         this.targetStreamsInfo = TypeConvert.jsonToObject(ctaInfo, ShoutoutTargetCallToAction.class);
     }
 

--- a/pubsub/src/test/java/com/github/twitch4j/pubsub/domain/CreateShoutoutDataTest.java
+++ b/pubsub/src/test/java/com/github/twitch4j/pubsub/domain/CreateShoutoutDataTest.java
@@ -1,0 +1,33 @@
+package com.github.twitch4j.pubsub.domain;
+
+import com.github.twitch4j.common.util.TypeConvert;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Tag("unittest")
+class CreateShoutoutDataTest {
+
+    @Test
+    @DisplayName("Deserialize Create Shoutout PubSub Data without CTA")
+    void deserializeSimple() {
+        String json = "{\"broadcasterUserID\":\"135074820\",\"targetUserID\":\"42941889\",\"targetLogin\":\"arnaudryku\"," +
+            "\"targetUserProfileImageURL\":\"https://static-cdn.jtvnw.net/jtv_user_pictures/71cc5b31-6cc1-4761-b7c4-7a1fccfaa56a-profile_image-%s.jpeg\"," +
+            "\"sourceUserID\":\"72738644\",\"sourceLogin\":\"diatorker\",\"shoutoutID\":\"d2a6ac54-1d8a-4e44-8b5d-f7c1ee4938e6\"," +
+            "\"targetUserDisplayName\":\"ArnaudRyku\",\"targetUserCTAInfo\":\"\",\"targetUserPrimaryColorHex\":\"000000\"}";
+
+        CreateShoutoutData data = assertDoesNotThrow(() -> TypeConvert.jsonToObject(json, CreateShoutoutData.class));
+        assertEquals("d2a6ac54-1d8a-4e44-8b5d-f7c1ee4938e6", data.getShoutoutId());
+        assertEquals("135074820", data.getBroadcasterId());
+        assertEquals("72738644", data.getModeratorId());
+        assertEquals("diatorker", data.getModeratorLogin());
+        assertEquals("42941889", data.getTargetId());
+        assertEquals("arnaudryku", data.getTargetLogin());
+        assertEquals("ArnaudRyku", data.getTargetDisplayName());
+        assertEquals("000000", data.getTargetUserColorHex());
+    }
+
+}

